### PR TITLE
Add .editorconfig for automatic editor configuration

### PR DESCRIPTION
**Pull request by vlsi**
The notable feature is automatic trim_trailing_whitespace=true configuration

### CODE REVIEW

**Comment by matzuk** [comment]
@RuslanMingaliev @eakurnikov do you have any suggestions for .editorconfig?

**Comment by vlsi** [comment]
I believe `.editorconfig` could have prevented https://github.com/KasperskyLab/Kaspresso/commit/fa35bb0dcb822bcc6312f73428b0fae677752db8

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+[*.md]
+# Trailing whitespace is important for Markdown
+trim_trailing_whitespace = false
+
+[*.{kt,kts}]
+max_line_length = 200
+
+[*.java]
+indent_size = 4
```
Comment (eakurnikov) [inline .editorconfig:17]: I think it should be used for [*]

**Review by eakurnikov — CHANGES_REQUESTED** [review]

**Comment by eakurnikov** [comment]
Thanks for your contribution, this is really relevant)

File: .editorconfig (hunk 1 above)
Comment (RuslanMingaliev) [reply .editorconfig:17]: I don't think so. We use 2 spaces for yaml configs, for example

Comment (vlsi) [reply .editorconfig:17]: Oh, whatever you like. Frankly speaking, I have never used Kaspresso :)
It was just recently mentioned when we were discussing a talk for https://heisenbug-piter.ru/ , and I noticed editorconfig might be helpful here

Comment (vlsi) [reply .editorconfig]: I've added `indent_size = 2` for yaml

**Comment by RuslanMingaliev** [comment]
LGTM

**Review by RuslanMingaliev — APPROVED** [review]

**Comment by matzuk** [comment]
@eakurnikov. Are your requested changes completed?

**Review by eakurnikov — APPROVED** [review]

**Pull request merged**